openshift/origin
# [wip] test-cmd cleanup

We've gotten to a point where our `test-cmd` suites are crusty. `admin.sh` is nearing 500 lines, `basicresources.sh` is almost at 350... running these tests is a real PITA.

In these commits, I do the following:
 - break up `test/cmd/admin.sh` into smaller bits
 - reconfigure some tests that were previously working only because of unrelated temporal dependencies
 - enforce `test-cmd` test bucket length
 - make tests more robust by not assuming they know their project's name
 - reconfigure `test-cmd` to continue trying the next suite if a test fails instead of failing once and for all

The intent here is:
 - make our tests easier to grok -- 500 lines of anything is going to be a mess
 - enforce better coding standards -- when you're adding tests 500 lines down in a file, you may not realize the correct place to be putting cleanup bits to make your tests re-entrant, for example
 - make our tests more addressable -- when I am adding tests, I want to be able to hot-loop as quickly as possible. currently, not only do I have to sit through a lengthy preamble (fixing in other pulls) but if my test is in `admin.sh` or the like I then also have to sit through 500 other tests running
 - make our tests more pointed -- there were a number of tests that had implicit and hidden temporal dependencies on other tests in entirely different suites. this should never happen and is very fragile

Review would be appreciated on the second commit, as it touches tests the most. The enforcement bit may be contentious, I don't feel super strong about it but I think it does add value.

@smarterclayton @liggitt @deads2k
